### PR TITLE
Split the lexer from the parser and put them in classes.

### PR DESCRIPTION
**Pull request by mbjorge**
In anticipation of the parser getting more complex with logic
for improved error messaging, split out the lexer to keep parser.py
cleaner.

Move the lexer and the parser into their own classes. This makes it
easier and cleaner to keep a references to lexer / parser instances
which are needed to do better error messaging when a syntax error comes
up.

Issue: #2
Test: stl/parser_test.py

### CODE REVIEW

**Comment by mbjorge** [comment]
This is just a refactor, should not have any behaviour changes. This will make doing the error messaging easier in a follow up CL

File: stl/lexer.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    To pass this into a Ply Yacc parser, pass it in using the .lexer propert
+    of an StlLexer instance:
+      my_lexer = StlLexer()
+      my_parser = ply.yacc.parser(lexer=my_lexer.lexer)
+
+    Args:
+      filename: The filename string to use in any error messaging.
+      kwargs: Forwarded to ply.lex.lex.
+    """
+    self.filename = filename
```
Comment (protobird-git) [inline stl/lexer.py]: At some point, we might need to distinguish private/protected members from public members. Private/protected members start with _ in google style.

Comment (mbjorge) [inline stl/lexer.py]: Done! Left the .lexer and .parser fields as is for now, since possibly will be using them as public

**Comment by protobird-git** [comment]
Would you please add class comments in a following CL which describe what public attributes are available?

**Pull request merged**